src/mon/ConfigMap.cc

```diff
@@ -166,9 +267,54 @@ ConfigMap::generate_entity_map(
   std::map<std::string,std::string,std::less<>> out;
   MaskedOption *prev = nullptr;
   for (auto s : sections) {
+    // apply profile content first
     for (auto& i : s.second->options) {
+      auto p = profiles.find(i.first);
+      if (p == profiles.end()) {
+	continue;
+      }
+
+      // match against crush location, class
       auto& o = i.second;
+      if (o.mask.device_class.size() &&
+	  o.mask.device_class != device_class) {
+	continue;
+      }
+      if (o.mask.location_type.size()) {
+	auto p = crush_location.find(o.mask.location_type);
+	if (p == crush_location.end() ||
+	    p->second != o.mask.location_value) {
+	  continue;
+	}
+      }
+      if (prev && prev->opt->name != i.first) {
+	prev = nullptr;
+      }
+      if (prev &&
+	  prev->get_precision(crush) < o.get_precision(crush)) {
+	continue;
+      }
+
+      // apply profile options
+      auto q = p->second.profile.find(o.raw_value);
+      if (q != p->second.profile.end()) {
+	for (auto& [opt, v] : q->second.options) {
+	  // only apply profile value if the value isn't already set
```

Comment:
does this mean the first profile to affect an option has precedence over subsequent profiles?

Reply:
that comment is obsolete, removing!